src/raster/r.viewshed.exposure/r.viewshed.exposure.py

```diff
@@ -0,0 +1,1409 @@
+    counter = 1
+
+    # Get variables out of global_vars dictionary
+    reg = global_vars["region"]
+    exp_range = global_vars["range"]
+    flagstring = global_vars["flagstring"]
+    r_dsm = global_vars["r_dsm"]
+    v_elevation = global_vars["observer_elevation"]
+    refr_coeff = global_vars["refr_coeff"]
+    memory = global_vars["memory"]
+    parametrise_viewshed = global_vars["param_viewshed"]
+    dsm_type = global_vars["dsm_type"]
+    b_1 = global_vars["b_1"]
+    cores = global_vars["cores"]
+    tempname = global_vars["tempname"]
+
+    # Create empty viewshed
+    np_cum = np.empty((reg.rows, reg.cols), dtype=np.single)
+    np_cum[:] = np.nan
+    tmp_vs = "{}_{}".format(tempname, os.getpid())
+
+    for target_pnt in target_pts_np:
+
+        # Display a progress info message
+        grass.percent(counter, len(target_pts_np), 1)
+        grass.verbose(
+            "Processing point {i} ({p:.1%})".format(
+                i=int(target_pnt[0]), p=counter / len(target_pts_np)
+            )
+        )
+
+        # Global coordinates and attributes of target point T
+        t_glob = target_pnt[1:]
+
+        # ======================================================================
+        # 1. Set local computational region: +/- exp_range from target point
+        # ======================================================================
+        # ensure that local region doesn't exceed global region
+        loc_reg_n = min(t_glob[1] + exp_range + reg.nsres / 2, reg.north)
+        loc_reg_s = max(t_glob[1] - exp_range - reg.nsres / 2, reg.south)
+        loc_reg_e = min(t_glob[0] + exp_range + reg.ewres / 2, reg.east)
+        loc_reg_w = max(t_glob[0] - exp_range - reg.ewres / 2, reg.west)
+
+        # pygrass sets region for pygrass tasks
+        lreg = deepcopy(reg)
+        lreg.set_bbox(Bbox(loc_reg_n, loc_reg_s, loc_reg_e, loc_reg_w))
+        lreg.set_raster_region()
+
+        # Create processing environment with region information
+        c_env = os.environ.copy()
+        c_env["GRASS_REGION"] = grass.region_env(
+            n=loc_reg_n, s=loc_reg_s, e=loc_reg_e, w=loc_reg_w
+        )
+
+        lreg_shape = [lreg.rows, lreg.cols]
+
+        # ======================================================================
+        # 2. Calculate binary viewshed and convert to numpy
+        # ======================================================================
+        vs = grass.pipe_command(
+            "r.viewshed",
+            flags="b" + flagstring,
+            input=r_dsm,
+            output=tmp_vs,
+            coordinates="{},{}".format(t_glob[0], t_glob[1]),
+            observer_elevation=0.0,
+            target_elevation=v_elevation,
+            max_distance=exp_range,
+            refraction_coeff=refr_coeff,
+            memory=int(round(memory / cores)),
+            quiet=True,
+            overwrite=True,
+            env=c_env,
+        )
+        vs.communicate()
+        clean_temp(vs.pid)
```

Comment:
Maybe comment on this, is this some workaround for a bug, the temp files should be cleaned automatically, no?

Reply:
It is indeed a workaround, see:
https://github.com/OSGeo/grass/issues/1436

Mentioned in comment in 294ce1d